crates/autopilot/src/driver_api.rs

```diff
@@ -40,9 +40,19 @@ impl Driver {
         &self,
         request: &settle::Request,
         timeout: std::time::Duration,
-    ) -> Result<settle::Response> {
-        self.request_response("settle", request, Some(timeout))
-            .await
+    ) -> Result<settle::Response, anyhow::Error> {
+        let result = self
+            .request_response::<settle::Result>("settle", request, Some(timeout))
+            .await?;
+
+        match result {
```

Comment:
Why not keep the original error type and access kind/description where we print?

Reply:
Not sure I completely understand the suggestion. We won't have any control on the type of errors the `driver` will be sending on `/settle`.
That's why we can't really assume we'll encounter the same error kinds we implemented in the reference driver.